python/lsst/ts/cbp/component.py

```diff
@@ -1,160 +1,399 @@
+
+    mask_select_status: float
+        This is the current value of the status of the mask selection encoder. If this value is non-zero, the encoder
+        has some kind of issue.
+
+    mask_rotate_status: float
+        This is the current value of the status of the mask rotation encoder. If this value is non-zero, the encoder
+        has some kind of issue.
+
+    focus_status: float
+        This is the current value of the status of the focus encoder. If this value is non-zero, the encoder has some
+        kind of issue.
+
+    Warnings
+    --------
+
+    The mask functionality is not quite done being implemented yet. This is because there are no physical masks yet and
+    therefore names are not final.
+
+    Notes
+    -----
+
+    The class uses the python socket module to build TCP/IP connections to the galil controller mounted onto CBP. The
+    underlying api is built on dmc,a language variant of c built by galil for controling motors and such items.
     """
     def __init__(self,address: str, port: int):
+        self.log = logging.getLogger(__name__)
         self.socket = None
         self.altitude = None
         self.azimuth = None
         self.mask = None
         self.mask_rotation = None
-        self.mask_dictionary = {}
+        self.mask_dictionary = bidict({1.:"Bob Hoskins 1",2.:"Bob Hoskins 2",3.:"Bob Hoskins 3",4.:"Bob Hoskins 4",5.:"Bob Hoskins 5",9.:"Unknown mask"})
+        self.mask_rotation_dictionary = {"Bob Hoskins 1":0,"Bob Hoskins 2":0,"Bob Hoskins 3":0,"Bob Hoskins 4":0,"Bob Hoskins 5":0}
         self.focus = None
         self._address = address
         self._port = port
+        self.connect()
         self.get_cbp_telemetry()
         self.panic_status = None
         self.check_panic_status()
+        self.auto_park = None
+        self.park = None
+        self.check_auto_park()
         self.azimuth_status = None
         self.altitude_status = None
         self.mask_select_status = None
         self.mask_rotate_status = None
         self.focus_status = None
         self.check_cbp_status()
+        self.log.info("CBP initialized")
 
     def connect(self):
+        """Creates a socket and connects to the CBP's static address and designated port.
+
+
+
+        Returns
+        -------
+
+        None
+
+        """
         self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.socket.settimeout(5)
         self.socket.connect((self._address, self._port))
+        self.log.debug("CBP connected to {0} on port {1}".format(self._address, self._port))
 
     def get_azimuth(self):
-        self.connect()
-        sleep(1)
+        """Gets azimuth value from azimuth encoder
+
+        Returns
+        -------
+
+        """
         self.socket.sendall("az=?\r".encode('ascii'))
-        self.azimuth = self.socket.recv(128).decode('ascii', 'ignore')
-        self.socket.close()
+        self.azimuth = float(self.socket.recv(128).decode('ascii', 'ignore').split("\r")[0])
```

Comment:
this line is repeated multiple times in a variety of methods. Please create a new method with just this functionality of parsing the returned string from the galil controller and returning the part you care about.